libraries/redcore/api/hal/hal.php

```diff
@@ -2300,7 +2329,7 @@ public function apiReadTypeItem(&$primaryKeys)
 		if (!empty($operations->read->item))
 		{
 			$dataGet = $this->triggerFunction('processPostData', $this->options->get('dataGet', array()), $operations->read->item);
-			$primaryKeysFromFields = RApiHalHelper::getPrimaryKeysFromFields($operations->read->item);
+			$primaryKeysFromFields = RApiHalHelper::getFieldsArray($operations->read->item);
```

Comment:
you are fetching all keys instead of just primary. This is not good. I would rather place a restriction in the editor to add fields that are not primary, or make them primary on save.